plugins/TestRunner/Commands/CheckDirectDependencyUse.php

```diff
@@ -0,0 +1,151 @@
+
+        $input = $this->getInput();
+        $plugin = $input->getOption('plugin');
+        $isGrepVendorFolder = $input->getOption('grep-vendor');
+
+        if (!empty($plugin)) {
+            $this->usesFoundList[$plugin] = [];
+        }
+
+        foreach ($psr4NamespacePrefixes as $prefix) {
+            $directUses = $this->grepForUses($prefix, 'psr4', $plugin, $isGrepVendorFolder);
+            if (!empty($directUses)) {
+                $this->reportDirectUses($prefix, $directUses, 'psr4');
+            }
+        }
+
+        foreach ($psr0Prefixes as $prefix) {
+            $directUses = $this->grepForUses($prefix, 'psr0', $plugin, $isGrepVendorFolder);
+            if (!empty($directUses)) {
+                $this->reportDirectUses($prefix, $directUses, 'psr0');
+            }
+        }
+
+        return self::SUCCESS;
+    }
+
+    private function getCoreDependencyNamespacePrefixes()
+    {
+        $psr4NamespacePrefixes = [];
+        $psr0Prefixes = [];
+
+        $coreComposerLock = PIWIK_INCLUDE_PATH . '/composer.lock';
+        $coreComposerLockContents = file_get_contents($coreComposerLock);
+        $coreComposerLockContents = json_decode($coreComposerLockContents, true);
+
+        foreach ($coreComposerLockContents['packages'] as $package) {
+            $psr4NamespacePrefixes = array_merge(
+                $psr4NamespacePrefixes,
+                array_keys($package['autoload']['psr-4'] ?? [])
+            );
+
+            $psr0Prefixes = array_merge(
+                $psr0Prefixes,
+                array_keys($package['autoload']['psr-0'] ?? [])
+            );
+        }
+
+        $psr4NamespacePrefixes = array_filter($psr4NamespacePrefixes);
+        $psr4NamespacePrefixes = array_unique($psr4NamespacePrefixes);
+
+        $psr0Prefixes = array_filter($psr0Prefixes);
+        $psr0Prefixes = array_unique($psr0Prefixes);
+
+        return [$psr4NamespacePrefixes, $psr0Prefixes];
+    }
+
+    private function grepForUses($prefix, $psrType, $plugin, $isGrepVendorFolder)
+    {
+        $uses = [];
+        $rgOutput = [];
+
+        if ($plugin) {
+            $plugin = '/' . $plugin;
+        }
+
+        $vendorScan = '--glob=\\!vendor';
+        if ($isGrepVendorFolder) {
+            $vendorScan = '';
+        }
+
+        if ($psrType === 'psr4') {
+            $prefix = rtrim($prefix, '\\');
+            $regex = ' \\\\?' . preg_quote($prefix) . '\\b';
+
+            $command = 'rg \'' . $regex . '\' --glob=*.php ' . $vendorScan . ' --json ' . PIWIK_INCLUDE_PATH . '/plugins' . $plugin;
+
+            exec($command, $rgOutput, $returnCode);
+        } else if ($psrType === 'psr0') {
+            $regex = '\\b' . preg_quote($prefix) . '_';
+            $command = 'rg \'' . $regex . '\' --glob=*.php ' . $vendorScan . ' --json ' . PIWIK_INCLUDE_PATH . '/plugins' . $plugin;
+
+            exec($command, $rgOutput, $returnCode);
+        }
```

Comment:
Can we simplify here? Looks like the same lines are repeated in both conditions.
Since the method is private it should be fine to not worry about passing in other PSR types and make psr0 the default as well.
```suggestion
        if ($psrType === 'psr4') {
            $prefix = rtrim($prefix, '\\');
            $regex = ' \\\\?' . preg_quote($prefix) . '\\b';
        } else {
            $regex = '\\b' . preg_quote($prefix) . '_';
        }
        
        $command = 'rg \'' . $regex . '\' --glob=*.php ' . $vendorScan . ' --json ' . PIWIK_INCLUDE_PATH . '/plugins' . $plugin;

        exec($command, $rgOutput, $returnCode);
```